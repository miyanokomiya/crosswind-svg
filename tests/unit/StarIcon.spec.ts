import { mount } from "@vue/test-utils";
import Target from "../../src/components/StarIcon.vue";

describe("StarIcon", () => {
  it("liked:false", () => {
    const wrapper = mount(Target, {
      propsData: { liked: false }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("liked:true", () => {
    const wrapper = mount(Target, {
      propsData: { liked: true }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
