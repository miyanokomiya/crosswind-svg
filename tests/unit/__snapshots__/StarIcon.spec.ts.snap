// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarIcon liked:false 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
  >
    <g
      fill="none"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%;"
    >
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
      />
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
        style="transform-origin: 50% 50%; transform: scaleX(-1);"
      />
    </g>
     
    <g
      fill=""
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%; transform: scale(0);"
    >
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
      />
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
        style="transform-origin: 50% 50%; transform: scaleX(-1);"
      />
    </g>
  </g>
</svg>
`;

exports[`StarIcon liked:true 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
  >
    <g
      fill="none"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%; transform: scale(0);"
    >
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
      />
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
        style="transform-origin: 50% 50%; transform: scaleX(-1);"
      />
    </g>
     
    <g
      fill=""
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 50% 50%;"
    >
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
      />
      <path
        d="M 32 14 38 26 50 26 41 35 44 47 32 40"
        style="transform-origin: 50% 50%; transform: scaleX(-1);"
      />
    </g>
  </g>
</svg>
`;
